x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/inference/InferenceProcessorTests.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.inference;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class InferenceProcessorTests extends ESTestCase {
+
+    public void testFactory() throws Exception {
+        String processorTag = randomAlphaOfLength(10);
+        
+        Model mockModel = mock(Model.class);
+        ModelLoader mockLoader = mock(ModelLoader.class);
+        when(mockLoader.load(eq("k2"), eq(processorTag), eq(false), any())).thenReturn(mockModel);
+
+        InferenceProcessor.Factory factory = new InferenceProcessor.Factory(Map.of("test", mockLoader));
+
+        Map<String, Object> config = new HashMap<>();
```

Comment:
Since you are using `Map.of` anyway in line 31, you could also use it here to make declaration of `config` more concise. The same in line 45.